File: python/rise/statistics/attack.py
class Attack(object):
    def __init__(self, accuracy, damage, name, defense=None, action_point=False):
        self.accuracy = accuracy
        self.action_point = action_point
        self.damage = damage
        self._defense = defense
        self.name = name

    @property
    def defense(self):
        return self._defense or "Armor"

    def __str__(self):
        return (
            f"Attack({self.name}: +{self.accuracy} ({self.damage})"
            + f"{' (AP)' if self.action_point else ''})"
        )

File: typescript/src/types/attack.ts
import { DicePool } from '../core_mechanics/dice_pool';
import type { PowerScaling } from '../core_mechanics/power_scaling';
import { DamageScaling } from '../core_mechanics/damage_scaling';
import { DamageDice } from './core';
export type AttackEffect =
  | { kind: 'Damage'; damage: DamageScaling }
  | { kind: 'Push'; distance: number }
  | { kind: 'Debuff'; debuff: string; duration: string }
  | { kind: 'Other'; text: string };

export interface Attack {
  accuracy: number;
  crit?: AttackEffect;
  defense: string;
  hit: AttackEffect;
  isMagical: boolean;
  isStrike: boolean;
  name: string;
  tags?: string[];
  targeting: string;
}
